pkg/transfer/image/imagestore_test.go

```diff
@@ -0,0 +1,265 @@
+			},
+			Annotations: []string{"OCI"},
+			ImageName:   "latest",
+			Images:      []string{"registry.test/base:latest", "registry.test/base@"},
+		},
+		{
+			Name: "NameSkipDigest",
+			ImageStore: &Store{
+				extraReferences: []Reference{
+					{
+						Name:            "registry.test/base",
+						IsPrefix:        true,
+						AddDigest:       true,
+						SkipNamedDigest: true,
+					},
+				},
+			},
+			Annotations: []string{"OCI"},
+			ImageName:   "latest",
+			Images:      []string{"registry.test/base:latest"},
+		},
+		{
+			Name: "OverwriteNameDigest",
+			ImageStore: &Store{
+				extraReferences: []Reference{
+					{
+						Name:           "base-name",
+						IsPrefix:       true,
+						AllowOverwrite: true,
+						AddDigest:      true,
+					},
+				},
+			},
+			Annotations: []string{"OCI", "containerd"},
+			ImageName:   "registry.test/base:latest",
+			Images:      []string{"registry.test/base:latest", "base-name@"},
+		},
+		{
+			Name: "OverwriteNameSkipDigest",
+			ImageStore: &Store{
+				extraReferences: []Reference{
+					{
+						Name:            "base-name",
+						IsPrefix:        true,
+						AllowOverwrite:  true,
+						AddDigest:       true,
+						SkipNamedDigest: true,
+					},
+				},
+			},
+			Annotations: []string{"OCI", "containerd"},
+			ImageName:   "registry.test/base:latest",
+			Images:      []string{"registry.test/base:latest"},
+		},
+		{
+			Name: "ReferenceNotFound",
+			ImageStore: &Store{
+				extraReferences: []Reference{
+					{
+						Name:     "registry.test/image",
+						IsPrefix: true,
+					},
+				},
+			},
+			Annotations: []string{"OCI", "containerd"},
+			ImageName:   "registry.test/base:latest",
+			Err:         errdefs.ErrNotFound,
+		},
+		{
+			Name:        "NoReference",
+			ImageStore:  &Store{},
+			Annotations: []string{"Annotation", "NoAnnotation"},
+			Err:         errdefs.ErrNotFound,
+		},
+		{
+			Name: "ImageName",
+			ImageStore: &Store{
+				imageName: "registry.test/index:latest",
+			},
+			Annotations: []string{"NoAnnotation"},
+			Images:      []string{"registry.test/index:latest"},
+		},
+	} {
+		testCase := testCase
+		for _, a := range testCase.Annotations {
+			name := testCase.Name + "_" + a
+			dgst := digest.Canonical.FromString(name)
+			desc := ocispec.Descriptor{
+				Digest:      dgst,
+				Annotations: map[string]string{},
+			}
+			expected := make([]string, len(testCase.Images))
+			for i, img := range testCase.Images {
+				if img[len(img)-1] == '@' {
+					img = img + dgst.String()
+				}
+				expected[i] = img
+			}
+			switch a {
+			case "containerd":
+				desc.Annotations["io.containerd.import.ref-source"] = "annotation"
+				desc.Annotations[images.AnnotationImageName] = testCase.ImageName
+			case "OCI":
+				desc.Annotations["io.containerd.import.ref-source"] = "annotation"
+				desc.Annotations[ocispec.AnnotationRefName] = testCase.ImageName
+			case "Annotation":
+				desc.Annotations["io.containerd.import.ref-source"] = "annotation"
+			}
```

Comment:
This switch is confusing.
What are "containerd", "OCI", "Annotation" in this context?

Reply:
I'll add a doc statement on the field for this